controllers/scheduler_test.go

```diff
@@ -334,212 +563,334 @@ var _ = Describe("Controllers/Scheduler", func() {
+						Expect(fakeK8sClient.List(context.Background(), &workPlacements)).To(Succeed())
 						Expect(workPlacements.Items).To(HaveLen(2))
 
 						devWorkPlacement := workPlacements.Items[0]
 						Expect(devWorkPlacement.Spec.TargetDestinationName).To(Equal(devDestination.Name))
 						Expect(devWorkPlacement.ObjectMeta.Labels["kratix.io/work"]).To(Equal(dependencyWorkForDev.Name))
+						Expect(devWorkPlacement.ObjectMeta.ResourceVersion < devWorkPlacementResourceVersionPreReconcile).To(BeTrue())
 
 						devWorkPlacement2 := workPlacements.Items[1]
 						Expect(devWorkPlacement2.Spec.TargetDestinationName).To(Equal(devDestination2.Name))
 						Expect(devWorkPlacement2.ObjectMeta.Labels["kratix.io/work"]).To(Equal(dependencyWorkForDev.Name))
+						Expect(devWorkPlacement2.ObjectMeta.ResourceVersion > devWorkPlacement2ResourceVersionPreReconcile).To(BeTrue())
 
-						//Remove finalizer
-						devWorkPlacement.Finalizers = nil
-						Expect(k8sClient.Update(context.Background(), &devWorkPlacement)).To(Succeed())
-						//manually delete workPlacement
-						Expect(k8sClient.Delete(context.Background(), &devWorkPlacement)).To(Succeed())
+						/*
+							TODO(question)
 
-						//re-reconcile
-						Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&dependencyWorkForDev), &dependencyWorkForDev))
-						_, err = scheduler.ReconcileWork(&dependencyWorkForDev)
-						Expect(err).ToNot(HaveOccurred())
+							I'm using ResourceVersion as a way to assert that the non-deleted
+							WorkPlacement is not recreated, but this might not be a good way to
+							go about this!
 
-						Expect(k8sClient.List(context.Background(), &workPlacements)).To(Succeed())
-						Expect(workPlacements.Items).To(HaveLen(2))
+							The ResourceVersion is updated because the OwnerReference.Controller
+							*bool is set to a different pointer-to-true in SetControllerReference.
+							This feels a bit iffy anyway, since it's the same value (just a different
+							pointer), so should we even be incrementing the ResourceVersion in this
+							case?
 
-						devWorkPlacement = workPlacements.Items[0]
-						Expect(devWorkPlacement.Spec.TargetDestinationName).To(Equal(devDestination.Name))
-						Expect(devWorkPlacement.ObjectMeta.Labels["kratix.io/work"]).To(Equal(dependencyWorkForDev.Name))
+							This means that we need to assert that the latest ResourceVersion is
+							_greater than_ the pre-reconcile ResourceVersion, but it feels like
+							we should be able to assert that the ResourceVersion is the _same_
+							(i.e. that devWorkPlacement2 doesn't get touched).
 
-						devWorkPlacement2 = workPlacements.Items[1]
-						Expect(devWorkPlacement2.Spec.TargetDestinationName).To(Equal(devDestination2.Name))
-						Expect(devWorkPlacement2.ObjectMeta.Labels["kratix.io/work"]).To(Equal(dependencyWorkForDev.Name))
+							Is there a better way to assert that devWorkPlacement2 is not recreated?
```

Comment:
you can use `creationTimestamp` instead; set it manually if its not created automatically by the fakeClient (or in the controller directly)

Reply:
> CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC. Populated by the system. Read-only. Null for lists.

Looks like we can't set `CreationTimestamp` 😢

Reply:
And because the fake client doesn't set it, I don't think we can use it to compare in the tests

Reply:
To get this PR out, I'm just going to assert that `devWorkPlacement2` is present for now, and add a TODO to ensure it's the same object!

Reply:
For posterity, here are the reasons why this is harder to test than it should be:
- the CreationTimestamp is not set by the fake client
- the UID is not set by the fake client
- the ResourceVersion _is_ set, but is updated on every reconcile because the status is updated* (we could assert that it's a known value here, but it might change if the code under test calls out to update the status any more than it currently does, so this might be flaky)

*This might be fake-client-specific - there are a couple of similar-ish Github issues, but I'm stopping investigation here! I did find out that it's definitely the status update in the Scheduler that's bumping the resource version, not the pointer-to-true candidate I thought about in a previous comment

Reply:
Also, we might not even _want_ to test this, since all we really care about (I think) is that the WorkPlacement for `devWorkPlacement2` exists, regardless of whether we recreate it or not